luda-editor/client/src/pages/graphic_asset_manage_page/cg_viewer/mod.rs

```diff
@@ -0,0 +1,267 @@
+mod part_picker;
+
+use self::part_picker::PartPicker;
+use crate::{color, components::cg_render::CgRender, storage::get_project_cg_thumbnail_image_url};
+use namui::prelude::*;
+use namui_prebuilt::{button, simple_rect, table::hooks::*, typography};
+use rpc::data::{ScreenCg, ScreenCgPart};
+use std::ops::Deref;
+
+const MODAL_MAX_WH: Wh<Px> = Wh {
+    width: px(1280.0),
+    height: px(720.0),
+};
+const MODAL_MIN_MARGIN: Px = px(16.0);
+const TITLE_BAR_HEIGHT: Px = px(48.0);
+const PICKER_WIDTH: Px = px(512.0);
+
+#[component]
+pub struct CgViewer<'a> {
+    pub wh: Wh<Px>,
+    /// NOTE: cg_viewer does not change the cg_file.
+    /// cg_file is used to init screen_cg then cg_viewer modifies screen_cg.
+    /// Change of cg_file means that the user opened a preview of another cg_file.
```

Comment:
That comment is likely to be confusing.
Actually, the CgViewer re-initializes screen_cg when the cg_file changes.

https://github.com/NamseEnt/namseent/pull/701/files/5265b58db9c77b605329fa91b5f09956e9584b2c#diff-4d01e51d02f8c07e212f498274e34bc1cd3be6e8a67fd0ea2e86fc90b5b51385R50-R56

Reply:
Then it should not implemented like this..!

Don't make this viewer complicated. Prevent side-effect and forcibly reduce the complexity of parameters.

"re-initialization" is not good idea at all. it will make a big problem and make problem complex.

Reply:
Think about current code. if cg_file changed, it won't update screen_cg directly. it will changed on next render by set_screen_cg. then there must be the gap between 'updating cg_file' and 'so update screen_cg by cg_file's change using set_screen_cg.'

Reply:
I simplified cg_viewer 9353f19.
Now cg_viewer just display screen_cg that served in prop, with no side-effects.